mlx/primitives.h

```diff
@@ -1856,4 +1856,18 @@ class SVD : public Primitive {
   void eval(const std::vector<array>& inputs, std::vector<array>& outputs);
 };
 
+/* Matrix inversion primitive. */
+class Inverse : public UnaryPrimitive {
+ public:
+  explicit Inverse(Stream stream) : UnaryPrimitive(stream){};
+
+  void eval_cpu(const std::vector<array>& inputs, array& output) override;
+  void eval_gpu(const std::vector<array>& inputs, array& output) override;
```

Comment:
Btw I feel like we should be implementing vmap for these primitives.. I didn't catch it for QR.. but it's actually not too bad to implement. You could move the vmapped axis to any axis other than the last two.

Reply:
Can I do it for both on a follow-up?

Reply:
Of course!